test/20.pass/test_make_heap_alloc.py

```diff
@@ -0,0 +1,37 @@
+import freetensor as ft
+
+
+def test_basic():
+    with ft.VarDef([("x", (), "int32", "cache", "cpu/heap"),
+                    ("i", (), "int32", "input", "cpu"),
+                    ("o", (), "int32", "output", "cpu")]) as (x, i, o):
+        with ft.VarDef("t", (), "int32", "cache", "cpu") as t:
+            t[()] = i[()] * 7
+            x[()] = t[()] + 2
+        o[()] = x[()] * 2
+        o[()] *= 2
+    ast = ft.pop_ast(verbose=True)
+    ast = ft.lower(ast, verbose=1)
+
+    with ft.VarDef([("i", (), "int32", "input", "cpu"),
+                    ("o", (), "int32", "output", "cpu")]) as (i, o):
+        o[()] = 28 * i[()] + 8
+    std = ft.pop_ast()
+
+    assert std.match(ast)
```

Comment:
Add `Alloc` and `Free` nodes to `include/debug/match_ast.h` to match them in a test.